src/coreclr/jit/codegencommon.cpp

```diff
@@ -627,11 +605,23 @@ regMaskTP Compiler::compHelperCallKillSet(CorInfoHelpFunc helper)
 {
     switch (helper)
     {
+        // Most of the helpers are written in C++ and C# and we can't make
+        // any additional assumptions beyond the standard ABI. However, some are written in raw assembly,
+        // so we can narrow down the kill sets.
+        //
+        // TODO-CQ: Inspect all asm helpers and narrow down the kill sets for them.
+        //
         case CORINFO_HELP_ASSIGN_REF:
         case CORINFO_HELP_CHECKED_ASSIGN_REF:
             return RBM_CALLEE_TRASH_WRITEBARRIER;
 
         case CORINFO_HELP_ASSIGN_BYREF:
+#ifdef TARGET_AMD64
+            if (IsTargetAbi(CORINFO_NATIVEAOT_ABI))
```

Comment:
There is no reason for native AOT to be different.

Reply:
Yeah it's not done yet, I am waiting for a green ci to make sure I did the unix version correctly. Do I need to copy the `WRITE_BARRIER_CHECK` pieces to NAOT version? it currently doesn't have it